RestApi/Sepes.RestApi/Services/AzureService.cs

```diff
@@ -96,17 +83,24 @@ public async Task DeleteSecurityGroup(string securityGroupName, string resourceG
         {
             await _azure.NetworkSecurityGroups.DeleteByResourceGroupAsync(resourceGroupName, securityGroupName);
         }
-        // ApplyNsg(...)
+        
         public async Task ApplySecurityGroup(string resourceGroupName, string securityGroupName, string subnetName, string networkName)
         {
             //Add the security group to a subnet.
-            await _azure.Networks.GetByResourceGroup(resourceGroupName, networkName).Update().UpdateSubnet(subnetName).WithExistingNetworkSecurityGroup(securityGroupName).Parent().ApplyAsync();
+            var nsg = _azure.NetworkSecurityGroups.GetByResourceGroup(resourceGroupName, securityGroupName);
```

Comment:
There was a function that would generate this subscribtion ID without relying on azure. GenerateNsgResourceID(string resourceGroupName, string nsgName)